packages/hardhat/test/poolTest.ts

```diff
@@ -47,60 +50,180 @@ describe("Pool", async () => {
   describe("Deposit", async () => {
 
     it("should allow liquidity provider to deposit", async () => {
+
       await sut.connect(lp1).deposit(utils.parseUnits("450"))
-      expect(await sut.wallet(lp1.address)).to.be.eq(utils.parseUnits("450"))
+
+      expect(await sut.balance()).to.be.eq(utils.parseUnits("450"))
+      expect(await sut.balanceOf(lp1.address)).to.be.eq(utils.parseUnits("450"))
```

Comment:
I'd add checks that also check the pool balance on the ERC20

Reply:
done